tracer/src/Datadog.Trace/Span.cs

```diff
@@ -399,8 +399,12 @@ internal void Finish(DateTimeOffset finishTimestamp)
         /// <param name="exception">The exception.</param>
         internal void SetException(Exception exception)
         {
-            Error = true;
-            SetExceptionTags(exception);
+            // We do not log BlockExceptions as errors
+            if (exception is not AppSec.BlockException)
```

Comment:
Perhaps log this? Do we expect this to happen as part of normal operation? If not, an error log, if so, then a debug log.

Reply:
In .net framework, this could happen under certain conditions and it would not be an error.